dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconMapper.java

```diff
@@ -60,13 +60,15 @@ public IconResponse from(Icon icon) {
           icon.getKeywords(),
           ci.getFileResourceUid(),
           ci.getCreatedByUserUid(),
-          getCustomIconReference(ci.getKey()));
+          getCustomIconReference(ci.getKey()),
+          icon.isCustom());
```

Comment:
here we know the icon is custom, can't we just return true here and false in the else block below?
this way we would save all the logic in the service and in the persistence layer